docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,294 @@
+
+@subsection guide_developer_styleguide_naming_files Files
+@brief Naming convention for files.
+
+@formattable{files}
+@formattableentry{<b>library</b>`_`<b>description</b>`.extension`,General library file,`core_mqtt_serializer.c`}
+@formattableentry{<b>library</b>`_internal.h`,Internal library header,`core_mqtt_internal.h`}
+@formattableentry{<b>library</b>`_`<b>demo</b>`_description.c`,Library demo source,`mqtt_demo_plaintext.c`}
+@formattableentry{<b>library</b>`_description_`<b>utest</b>`.c`<br><b>library</b>`_description_`<b>system_test</b>`.c`,Library test source,`core_mqtt_utest.c`<br>`mqtt_system_test.c`}
+
+File names contain only lowercase letters and underscores. All file names should be named according to their purpose. For example:
+- `core_mqtt_state.c`: A file in the MQTT library that handles state of MQTT PUBLISH packet deliveries.
+- `mqtt_demo_plaintext.c`: A file in the demos for the MQTT library that communicates over a plaintext channel.
+- `core_mqtt_utest.c`: A file in the Tests for the MQTT library. Since the tests currently only run on POSIX systems, test file names do not use the `_posix` suffix.
+
+Library file names should use one or two words to describe the functions implemented in that file. For example:
+- `core_mqtt_serializer.c`: Implements the MQTT library's packet serialization and deserialization functions.
+- `clock_posix.c`: Implements the platform clock component for POSIX systems.
+
+Declarations of internal functions, structures, macros, etc. of a library should be placed in a header file with an `_internal` suffix. The `_internal` header file should go in the `source/include/private` directory. For example:
+- `core_mqtt_internal.h`: Declares the MQTT library's internal functions, structures, macros, etc.
+
+<p>File names for demos should begin with `library_demo_`. 
+Unit tests must start with `library_` and end with `_utest`. 
+System tests must start with `library_` and end with `_system_test`. 
+The names should then specify the library being demoed or tested. 
+For example, the files names of the MQTT library's demos start with `mqtt_demo_`. 
+Additionally, test file names should describe what tests are implemented in the file, 
+such as `mqtt_demo_mutual_auth.c` for a file containing a demo that mutually 
+authenticates over TLS with an MQTT broker.</p>
+
+@subsection guide_developer_styleguide_naming_functions Functions (and function-like macros)
+@brief Naming convention of functions and function-like macros.
+
+@formattable{functions}
+@formattableentry{<b>Library</b>`_`<b>Description</b>,Externally-visible library function,`MQTT_Publish`}
+@formattableentry{<b>description</b>,`static` function (never uses `Aws` prefix),`sendPublish`<br>`eventCallback`}
+@formattableentry{`test_`<b>Library</b>`_`<b>accessedFunction_Descrption</b>,Test access function,`test_MQTT_ProcessLoop_Timer_Overflow`}
```

Comment:
Description. Also, it looks like the description is optional? I don't think all test functions have one
```suggestion
@formattableentry{`test_`<b>Library</b>`_`<b>accessedFunction_</b><i>Description</i>,Test access function,`test_MQTT_ProcessLoop_Timer_Overflow`}
```

Reply:
Putting the italic before the underscore

Reply:
Actually, I'm removing the italic because it would be best to have a description for each test case